### ISSUE TITLE
Running without cluster permissions

### ISSUE THREAD

**Issue by jackivanov** [haproxytech/kubernetes-ingress#359]
Hello,

It would be nice to have an option to run the controller in the namespaced scope, without permissions at the cluster level (ClusterRole and ClusterRoleBinding). Role and RoleBinding can help with that.

Thanks

**Comment by dkorunic** [comment]
@jackivanov Have you taken a look at our [Ingress Controller Helm Chart](https://github.com/haproxytech/helm-charts/tree/main/kubernetes-ingress)?

**Comment by jackivanov** [comment]
@dkorunic Yes, and I can see it's using **ClusterRole** if `rbac.enabled`. Would that worked in a namespaced scope if I just disable rbac? I haven't found any clarification regarding that

**Comment by jackivanov** [comment]
Without rbacn enabled, the controller is not able to start

{code}
... truncated
E0818 10:43:08.486789       6 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.0/tools/cache/reflector.go:156: Failed to watch *v1.Namespace: failed to list *v1.Namespace: namespaces is forbidden: User "system:serviceaccount:workshop:upwork-haproxy-kubernetes-ingress" cannot list resource "namespaces" in API group "" at the cluster scope
2021/08/18 10:43:14 INFO    controller.go:120 Stopping Ingress Controller
2021/08/18 10:43:14 ERROR   controller.go:121 waitid: no child processes
{code}

Basically, my use case is that I want to run a controller in a specific namespaces without permissions at the cluster level in a multi-tenancy cluster where admin permissions are not given to consumers by obvious reasons.

**Comment by Mo3m3n** [comment]
This should be possible by converting  our ClusterRole and ClusterRoleBinding that you can find in the [deploy dir](https://github.com/haproxytech/kubernetes-ingress/blob/master/deploy/haproxy-ingress.yaml) to Role and RoleBinding resources

**Comment by sasati-tibco** [comment]
@jackivanov did you found the workaround ? The controller goes into crashloopbackoff by converting [clusterrole](https://github.com/haproxytech/helm-charts/blob/main/kubernetes-ingress/templates/clusterrole.yaml)  and [clusterrolebinding](https://github.com/haproxytech/helm-charts/blob/main/kubernetes-ingress/templates/clusterrolebinding.yaml) to role and rolebinding. 
@Mo3m3n ^^

**Comment by sasati-tibco** [comment]
@jackivanov just curious what was the solution that you used for this usecase.
